contracts/main/bridge/FathomBridge.sol

```diff
@@ -0,0 +1,159 @@
+                accessControlConfig.hasRole(accessControlConfig.SHOW_STOPPER_ROLE(), msg.sender),
+            "!(ownerRole or showStopperRole)"
+        );
+        _;
+    }
+
+    modifier onlyWhitelisted() {
+        if (!whitelisted[msg.sender] && !isDecentralizedMode) {
+            revert("FathomBridge/not-whitelisted");
+        }
+        _;
+    }
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    // --- Init ---
+
+    function initialize(IInitializerSender _initializerLib, address _stablecoin, address _accessControlConfig) external initializer {
+        _zeroAddressCheck(address(_initializerLib));
+        _zeroAddressCheck(_stablecoin);
+        _zeroAddressCheck(_accessControlConfig);
+        stablecoin = _stablecoin;
+        accessControlConfig = IAccessControlConfig(_accessControlConfig);
+        live = 1;
+        whitelisted[msg.sender] = true;
+        _asterizm_initialize(_initializerLib, true, false);
+    }
+
+    // --- Whitelisting ---
+
+    function addToWhitelist(address _usr) external onlyOwnerOrGov {
+        _zeroAddressCheck(_usr);
+        whitelisted[_usr] = true;
+        emit LogAddToWhitelist(_usr);
+    }
+
+    function removeFromWhitelist(address _usr) external onlyOwnerOrGov {
+        _zeroAddressCheck(_usr);
+        whitelisted[_usr] = false;
+        emit LogRemoveFromWhitelist(_usr);
+    }
+
+    // --- Administration ---
+
+    function setDecentralizedMode(bool _isOn) external onlyOwnerOrGov {
+        isDecentralizedMode = _isOn;
+        emit LogSetDecentralizedMode(_isOn);
+    }
+
+    function setFee(uint256 _newFee) external onlyOwnerOrGov {
+        fixedBridgeFee = _newFee;
+        emit LogSetFee(_newFee);
+    }
+
+    function withdrawFees(address _to) external onlyOwnerOrGov {
+        _zeroAddressCheck(_to);
+        stablecoin.safeTransfer(_to, stablecoin.balanceOf(address(this)));
+        emit LogWithdrawFees(msg.sender, _to, stablecoin.balanceOf(address(this)));
+    }
+
+    /// Cross-chain transfer
+    /// @notice This function is used to transfer FXD from the source chain to the destination chain.
+    /// It works only when the off chain client module is up and running to listen to this function's event.
+    /// works only when the sender is whitelisted or in decentralized mode
+    /// @param _dstChainId uint64  Destination chain ID
+    /// @param _to address  To address
+    /// @param _amount uint  Amount
+    function crossChainTransfer(uint64 _dstChainId, address _to, uint _amount) external onlyWhitelisted nonReentrant{
+        require(live == 1, "FathomBridge/not-live");
+        require(_amount > fixedBridgeFee, "FathomBridge/amount-less-than-fee");
+        _zeroAddressCheck(_to);
+
+	    stablecoin.safeTransferFrom(msg.sender, address(this), _amount);
+        uint256 _actualTransferAmount = fixedBridgeFee != 0 ? _amount - fixedBridgeFee : _amount;
+        IStablecoin(stablecoin).burn(msg.sender, _actualTransferAmount);
```

Comment:
this part doesn't look right to me. So `msg.sender` wants to sent some `_amount`. And here we first transfer this balance to the bridge, and than burn `amount-fee` from his balance. 
Lets say _amount = 1000 and fee = 10: in this case msg.sender will spend 1990 fxd instead of 1000 to transfer 990 fxd (1000 transferred and 990 burned). I thin this should look like this:
```
uint256 actualTransferAmount = _amount;
if (fixedBridgeFee > 0) {
stablecoin.safeTransferFrom(msg.sender, address(this), fixedBridgeFee);
actualTransferAmount -= _amount;
}
IStablecoin(stablecoin).burn(msg.sender, _actualTransferAmount);
```
or 
```
stablecoin.safeTransferFrom(msg.sender, address(this), _amount);
uint256 _actualTransferAmount = fixedBridgeFee != 0 ? _amount - fixedBridgeFee : _amount;
IStablecoin(stablecoin).burn(address(this), _actualTransferAmount);
```

Reply:
That's right. I should have changed msg.sender to address(this) when burning. Thanks.